### PR TITLE
Fix ListBuffer.insertAfter, used in patchInPlace

### LINKED ISSUE #12796: ListBuffer.patchInPlace does not update last0 field, leaving the buffer in a corrupted state
## Reproduction steps

Scala version: 2.13.10

```scala
    import collection.mutable.ListBuffer
    val buffer = ListBuffer(1)
    val patch = Seq(2)
    buffer.patchInPlace(1, patch, 2)
    println(buffer.last)
```

## Problem
Prints `1`

### PR DESCRIPTION
**Pull request by lrytz**
If the new elements are added to the end, `last0` needs to be updated.

`insertAfter` is used in `patchInPlace` and `insertAll`, but the bug didn't manifest for `insertAll` because it delegates to `addAll` when adding to the end.

Fixes https://github.com/scala/bug/issues/12796

### CODE REVIEW

**Comment by lrytz** [comment]
@SethTisue include in 2.13.11?

**Review by NthPortal — APPROVED** [review]

**Comment by SethTisue** [comment]
> include in 2.13.11?

Um, I'm torn. On the one hand, bugs in collections methods are high priority.

But on the other hand: the bug is highly specific to a particular collection and a particular not-super-commonly-used method on that collection. And, timingwise, 3.3.0 is out; we've already announced a 2.13.11 release candidate and we're 5 days into the 7 day countdown; and throwing in bugfixes at the last minute is admittedly always tempting, but risky. Because 3.3.0 itself was delayed, 2.12.18 and 2.13.11 are really overdue at this point. I suggest, unhappily, that we hold this one for 2.13.12.

**Comment by lrytz** [comment]
I'm generally very cautious, but considering merging in this case because it's in the standard library and it's very, very low risk. I'm really confident about the fix, the only danger I can imagine is that we were incorrectly relying on the bug somewhere, but that's also very unlikely.

Timing-wise, would it have to delay the release?

**Comment by SethTisue** [comment]
> Timing-wise, would it have to delay the release?

Yes — that is, unless you're also willing to skip announcing a new release candidate and resetting the 7 day waiting period.

**Comment by lrytz** [comment]
Right, skipping that would go too far I agree. So I guess the question is whether it's worth delaying the release. Hard to say, the report could have come 3 days later and we wouldn't be thinking about it. OTOH, it came now and we could still include the fix... ⚖️ If noone else speaks up, I'd say it's up to you as the release manager.

**Comment by som-snytt** [comment]
You can always say "just one more fix."

(This fix has minimal return, so isn't worth any risk. Also, I thought the reporter said they would provide a comprehensive fix?)

I'd suggest planning a shorter cycle for what is it 2.13.12.

**Comment by SethTisue** [comment]
> I'd suggest planning a shorter cycle for what is it 2.13.12.

Yes, let's shoot for 3 months or so. This time there won't be a Scala 3 minor release affecting the timing.

**Comment by som-snytt** [comment]
@SethTisue Happy June! That sounds perfect, as Lukas will just be returning from summer vacation. (Just guessing.)

**Pull request merged**
